src/librustc/traits/fulfill.rs

```diff
@@ -534,8 +534,7 @@ fn process_predicate<'a, 'gcx, 'tcx>(
                             } else {
                                 Err(CodeSelectionError(ConstEvalFailure(ConstEvalErr {
                                     span: obligation.cause.span,
-                                    kind: ErrKind::UnimplementedConstVal("could not resolve")
-                                        .into(),
+                                    kind: ErrKind::CouldNotResolve.into(),
```

Comment:
Wait, what error does miri produce from its own failed `Instance::resolve`?

Reply:
TypeckError: https://github.com/rust-lang/rust/blob/master/src/librustc_mir/interpret/eval_context.rs#L283

Reply:
That's... disappointing. I believe failing to resolve means "still too polymorphic", does miri not have any general errors for that, short of the layout ones?

Reply:
No, but I will add one when refactoring away the old ctfe error enum